alica_engine/src/engine/TeamObserver.cpp

```diff
@@ -297,41 +302,41 @@ std::unique_ptr<SimplePlanTree> TeamObserver::sptFromMessage(AgentId agentId, co
     SimplePlanTree* curParent = nullptr;
     SimplePlanTree* cur = root.get();
 
-    for (int i = 2; i < static_cast<int>(ids.size());) {
-        const int64_t id = ids[i];
-        if (id == -1) {
-            curParent = cur;
-            cur = nullptr;
-            ++i;
-        } else if (id == -2) {
+    // start at 2, because we used the first 2 ids for creating the root node
+    for (int i = 2; i < static_cast<int>(ids.size()); ) {
+        const int64_t currentEpId = ids[i];
+        if (currentEpId == -1) {
+            // go up one level
             cur = curParent;
             if (cur == nullptr) {
-                ALICA_WARNING_MSG("TO: Malformed SptMessage from " << agentId);
-                return nullptr;
+                // went up one level from root node, finished parsing
+                break;
             }
             curParent = cur->getParent();
-            ++i;
+            continue;
+        }      
+        // Add node as child to old current node
+        const int64_t currentStateId = ids[i + 1];
+        if (i + 1 == static_cast<int>(ids.size())) {
```

Comment:
this check should happen before the previous line

Reply:
@bjoernschroeder Can you address this one?